Midterm

Kavignon opened the pull request:
Content from sprint 0 to 3 should be completed by now.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/81)

<!-- Reviewable:end -->

Commit:
Dead letters are supported.

Commit:
SRS document almost completed. Missing a few use cases

Commit:
Finish modeling game elements for the brainy characters

Commit:
Finish modeling basic game logic to test how a character would learn how to play with available moves and a move range = 1.

Commit:
Up and running randomiser in main app. The chaacter moves and selects action randomly

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/81" target="_blank"><img src="https://app.review.ninja/66727011/pull/81/badge" alt="ReviewNinja"/>

Commit:
Solve issue: Character only moving in a 2 square radius at any given direction

Commit:
Fix issue why the score was staying at 0...

Commit:
#14  -  Implementing a basic brain which will learn from experience based on the SARSA algorithm

Commit:
The first stage of the primitive brain is complete
Because of the many option in the cells and possible moves, I will need to reduce them, it causes logical error but the brain does learn after all.

Commit:
Added the test project
Will property test the requirements. Those properties should be true for any given implementations
Will test the Akka architecture

Commit:
Added test files for the first step of the continuous integration architecture with Travis CI

Commit:
#4 - Added the A* pathfinding algorithm thanks to @jdoig

Commit:
Added unit test for GameElements

Commit:
Set up the travis CI file for continuous integration

Commit:
Finished the SRS document

Commit:
Présentation Générale complétée

Commit:
Finished Gameplay in game mechanics for the Game Design Document

Commit:
The game mechanics section in the Game design document is completed.

Commit:
The game design document is completed.

Commit:
The board is updated each time that a cell is consummed
Simplified game states int values going from -200 (VERY BAD) to +250 ( VERY GOOD)
Missing the Q-learning in the Advanced Brain module.

Commit:
The brain manager has the implementation for the primitive brain which basic learns how the environement works and what actions are available
With Q-learning, the advanced brain is able to plan ahead and not act for a direct gain, it also leaves room for exploration and learn what are truly the best options available

Kavignon commented:
The continuous integration will have to wait for later.
